src/Microsoft.VisualStudio.ProjectSystem.Managed/Threading/Tasks/CancellationSeries.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+
+namespace Microsoft.VisualStudio.Threading.Tasks
+{
+    /// <summary>
+    /// Produces a series of <see cref="CancellationToken"/> objects such that requesting a new token
+    /// causes the previously issued token to be cancelled.
+    /// </summary>
+    /// <remarks>
+    /// <para>Consuming code is responsible for managing overlapping asynchronous operations.</para>
+    /// <para>This class has a lock-free implementation to minimise latency and contention.</para>
+    /// </remarks>
+    public sealed class CancellationSeries : IDisposable
+    {
+        private CancellationTokenSource _cts = new CancellationTokenSource();
+
+        private readonly CancellationToken[] _superToken;
```

Comment:
Why is this an array? It never has more than one element.

Reply:
It's passed to a `params` method. This avoids allocating on each call.

Reply:
There's an overload of CreateLinkedTokenSource that accepts two tokens... instead of using an array here, you could just pass this "super token" as one of the inputs and pass `default`/`CancellationToken.None` as the other.